mily/runengine.py

```diff
@@ -0,0 +1,305 @@
+import logging
+from bluesky import RunEngine
+from bluesky.utils import RunEngineInterrupted, install_qt_kicker
+from qtpy.QtWidgets import QVBoxLayout, QLabel, QWidget, QButtonGroup, QPushButton, QDialogButtonBox
+from qtpy.QtCore import Signal, Slot
+from . import threads
+from bluesky import RunEngine, Msg
+from qtpy.QtCore import QObject, Signal
+from bluesky.preprocessors import subs_wrapper
+
+logger = logging.getLogger(__name__)
+
+# TODO: Allow queueing plans
+# TODO: Application logging
+# TODO: Capture state callbacks
+# TODO: Add convenient gui-thread targeted callback mechanism
+
+
+class QRunEngine(QObject):
+    sigDocumentYield = Signal(str, dict)
+    sigAbort = Signal()  # TODO: wireup me
+    sigException = Signal()
+    sigFinish = Signal()
+    sigStart = Signal()
+    sigPause = Signal()
+    sigResume = Signal()
+
+    def __init__(self, runengine=None, **kwargs):
+        super(QRunEngine, self).__init__()
+
+        self.RE = runengine or RunEngine(context_managers=[], **kwargs)
```

Comment:
Given that we need to create the RE with just the right settings (no context managers, probably a different `during_task`, and maybe using qthreads instead of python threads to push the asyncio event loop to the background, maybe with qumash to embed the asyncio event loop in the qt event loop (!!)) I don't think we should allow the user to pass in an existing `RunEngine`